railties/lib/rails/commands/credentials/credentials_command.rb

```diff
@@ -115,6 +117,19 @@ def extract_environment_from_path(path)
           available_environments.find { |env| path.include? env } if path.end_with?(".yml.enc")
         end
 
+        # Check for a mismatch between Rails::Command.environment and option set by --environment flag.
+        # This warning should only occur in non-development environments to avoid confusing the user
+        # when trying to perform this action against different environment modes.
+        def warn_if_environment_mismatch(method_name: :show)
+          if Rails::Command.environment != "development" && Rails::Command.environment != options[:environment]
+            say <<~WARN
+              You are accessing #{options[:environment]} credentials from the #{Rails::Command.environment} environment.
+              Did you mean to run `rails credentials:#{method_name} --environment #{Rails::Command.environment}`?
+              For more information try `rails credentials:help`.
```

Comment:
Prefer using `bin/rails` here, as it’ll be copy-pasteable and work everywhere:

```suggestion
              Did you mean to run `bin/rails credentials:#{method_name} --environment #{Rails::Command.environment}`?
              For more information try `bin/rails credentials:help`.
```

Reply:
Noted. I'll follow the same from now on. :))